src/scheduler.rs

```diff
@@ -701,7 +701,7 @@ async fn smtp_loop(
                     "smtp has messages to retry, planning to retry {} seconds later", timeout
                 );
                 let duration = std::time::Duration::from_secs(timeout);
-                tokio::time::timeout(duration, async {
+                tools::timeout(duration, async {
```

Comment:
@Simon-Laux and me assume that this was the offending line:

At 

https://github.com/deltachat/deltachat-core-rust/blob/a2495716b6c43a0c6ea82bd23c4c71a2463be712/src/scheduler.rs#L668

we multiply the timeout by 3 every time a message can't be sent, whether . So, if the user sends 30 messages, `idle_interrupt_receiver` receives an interrupt every time, the message fails to be sent every time, and the timeout is multiplied by 3 every time.